### PR TITLE
Gracefully exit when nonexistent or invalid colorscheme is supplied

### PR DESCRIPTION
**Pull request by sander2324**
ytop currently panics when a nonexistent or invalid colorscheme is supplied. With this PR, it will now print a clearer error and gracefully exit.

### CODE REVIEW

File: src/colorscheme.rs [hunk 1]
```diff
@@ -118,27 +119,43 @@ fn convert_color(raw: i64) -> Color {
...
-fn parse_colorscheme(config_folder: &Path, colorscheme: &Colorschemes) -> ColorschemeRaw {
+fn get_colorscheme_config(config_folder: &Path, colorscheme: &Colorschemes) -> String {
 	match colorscheme {
-		Colorschemes::Custom(name) => serde_json::from_str(
-			&fs::read_to_string(config_folder.join(name).with_extension("json")).unwrap(),
-		)
-		.unwrap(),
+		Colorschemes::Custom(name) => {
+			let config_file = config_folder.join(name).with_extension("json");
+			match fs::read_to_string(&config_file) {
```
Comment (cjbassi) [inline src/colorscheme.rs]: This can be replaced with `.unwrap_or_else`.

File: src/colorscheme.rs [hunk 2]
```diff
@@ -118,27 +119,43 @@ fn convert_color(raw: i64) -> Color {
 	}
 }
 
-fn parse_colorscheme(config_folder: &Path, colorscheme: &Colorschemes) -> ColorschemeRaw {
+fn get_colorscheme_config(config_folder: &Path, colorscheme: &Colorschemes) -> String {
 	match colorscheme {
-		Colorschemes::Custom(name) => serde_json::from_str(
-			&fs::read_to_string(config_folder.join(name).with_extension("json")).unwrap(),
```
Comment (cjbassi) [inline src/colorscheme.rs:124]: I think a cleaner way to implement this pr would be to just change these unwraps to exit gracefully. I don't think we need to change any of the functions.

**Review by cjbassi — CHANGES_REQUESTED** [review]
Thanks for the pr. I made a couple of suggestions if you could fix them and then i can merge this.

File: src/colorscheme.rs (hunk 2 above)
Comment (sander2324) [reply src/colorscheme.rs:124]: I've updated the PR and the code should be much cleaner now.

**Review by cjbassi — APPROVED** [review]

**Comment by cjbassi** [comment]
Nice, this looks great. Thanks for these changes!

**Pull request merged**